Source/Common/CriticalSection.cpp

```diff
@@ -1,16 +1,28 @@
 #include "stdafx.h"
+#ifdef _WIN32
 #include <windows.h>
+#else
+#include <mutex>
+#endif
 
 CriticalSection::CriticalSection()
 {
-	m_cs = new CRITICAL_SECTION;
-	::InitializeCriticalSection((CRITICAL_SECTION *)m_cs);
+#ifdef _WIN32
+    m_cs = new CRITICAL_SECTION;
+    ::InitializeCriticalSection((CRITICAL_SECTION *)m_cs);
+#else
+    m_cs = static_cast<void*>(new std::mutex);
```

Comment:
For example: here.
Visual Studio 2014 (at least that version) can do static_cast.